### DISCUSSION TITLE
OpenIPSL v2.0.0

### DISCUSSION THREAD

**Discussion by dietmarw** [OpenIPSL/OpenIPSL#298] [Announcements]
## What's Changed
This new release version 2.0.0 of **OpenIPSL** is a **non-backwards compatible** release based on the [Modelica Standard Library version 3.2.3](https://github.com/modelica/ModelicaStandardLibrary/releases/tag/v3.2.3%2Bbuild.4). 
A conversion script is provided for user libraries using **OpenIPSL** version 1.5.0 and older. *(Note: Models that extended from any of the old Examples models will have to be converted manually by the user.)* 
### Highlights
* Changed the license to the [3-Clause BSD License](https://opensource.org/licenses/BSD-3-Clause).
* Use of SI or per-unit units **only** in quantities (i.e., no more customs scaled kilo or Mega units).
* The old `Examples` models have been moved under `Tests`.
* The old external `Application Examples` package has now been incorporated as `Examples` and is part of this library. 
### Changes in detail
This release consists of [39 closed issues](../../issues?q=is%3Aissue+milestone%3A2.0.0+is%3Aclosed) and [61 merged pull requests](../../pulls?q=is%3Apr+milestone%3A2.0.0+is%3Aclosed): 
* fix typo in DerivativeLag by @pierre-haessig in https://github.com/OpenIPSL/OpenIPSL/pull/156
* Update to new version of journal paper by @lvanfretti in https://github.com/OpenIPSL/OpenIPSL/pull/159
* This fixes the description string of the sub-transient time constants by @dietmarw in https://github.com/OpenIPSL/OpenIPSL/pull/163
* add ControlledBreaker model under Events sub-package by @BiswarupM in https://github.com/OpenIPSL/OpenIPSL/pull/169
* Update README.md by @lvanfretti in https://github.com/OpenIPSL/OpenIPSL/pull/171
* Fix error in Tutorial: by @MaximeBaudette in https://github.com/OpenIPSL/OpenIPSL/pull/161
* (doc) Small update in publication list by @MaximeBaudette in https://github.com/OpenIPSL/OpenIPSL/pull/158
* Updating the KundurSMIB package to make use of the Generator interface by @dietmarw in https://github.com/OpenIPSL/OpenIPSL/pull/175
* Rebased PlantModel branch (replaces PR #157) by @dietmarw in https://github.com/OpenIPSL/OpenIPSL/pull/177
* Fixes #172 by removing duplicate implementations (with conversion) by @dietmarw in https://github.com/OpenIPSL/OpenIPSL/pull/178
* Clean implementation of the two and three winding transformer by @dietmarw in https://github.com/OpenIPSL/OpenIPSL/pull/180
* N44 update by @marcelofcastro in https://github.com/OpenIPSL/OpenIPSL/pull/181
* Fix defaultComponentPrefixes annotation by @tbeu in https://github.com/OpenIPSL/OpenIPSL/pull/184
* This fixes #112 and beyond by now introducing proper SI unit types by @dietmarw in https://github.com/OpenIPSL/OpenIPSL/pull/183
* Fixes #174 by providing a non-zero start value for ii and ir by @dietmarw in https://github.com/OpenIPSL/OpenIPSL/pull/190
* Fixes #93 by removing superfluous shunt models by @dietmarw in https://github.com/OpenIPSL/OpenIPSL/pull/191
* Adding discrete ULTC (closes #189) by @dietmarw in https://github.com/OpenIPSL/OpenIPSL/pull/192
* Machine base power rating change by @ManuelNvro in https://github.com/OpenIPSL/OpenIPSL/pull/196
* Removing the infinite bus from the example by @dietmarw in https://github.com/OpenIPSL/OpenIPSL/pull/203
* Remove hard coded and doubly defined start values for w and delta by @dietmarw in https://github.com/OpenIPSL/OpenIPSL/pull/204
* Add new ThreePhase package by @MaximeBaudette in https://github.com/OpenIPSL/OpenIPSL/pull/198
* Fix exciters by @dietmarw in https://github.com/OpenIPSL/OpenIPSL/pull/209
* Fixing the machine models by @dietmarw in https://github.com/OpenIPSL/OpenIPSL/pull/211
* Fixing governor models  by @dietmarw in https://github.com/OpenIPSL/OpenIPSL/pull/215
* Fix wind models by @dietmarw in https://github.com/OpenIPSL/OpenIPSL/pull/216
* Fixing PSS models by @dietmarw in https://github.com/OpenIPSL/OpenIPSL/pull/214
* Fix Banks models by @dietmarw in https://github.com/OpenIPSL/OpenIPSL/pull/213
* Fix exciter models by @dietmarw in https://github.com/OpenIPSL/OpenIPSL/pull/212
* Sorting the examples and excitation models alphabetical by @dietmarw in https://github.com/OpenIPSL/OpenIPSL/pull/222
* Remove SMIBpartial2 and instead update SMIBpartial by @dietmarw in https://github.com/OpenIPSL/OpenIPSL/pull/225
* Mark verified models via a distinctive icon by @dietmarw in https://github.com/OpenIPSL/OpenIPSL/pull/228
* Correct calculation of I_T in ESSTB4 by @dietmarw in https://github.com/OpenIPSL/OpenIPSL/pull/230
* Adds documentation of verification  by @dietmarw in https://github.com/OpenIPSL/OpenIPSL/pull/218
* Fix Terminator dimension warning by @dietmarw in https://github.com/OpenIPSL/OpenIPSL/pull/231
* Include PSAT PSS and load models from FOSSEE1 branch by @dietmarw in https://github.com/OpenIPSL/OpenIPSL/pull/232
* Explains the PSAT AVR models mix up by @dietmarw in https://github.com/OpenIPSL/OpenIPSL/pull/233
* Fix the CI setup by @dietmarw in https://github.com/OpenIPSL/OpenIPSL/pull/224
* Fix HTML syntax errors and also add package icons.  by @dietmarw in https://github.com/OpenIPSL/OpenIPSL/pull/235
* Better implementation of PSTransformer by @dietmarw in https://github.com/OpenIPSL/OpenIPSL/pull/236
* Sort classes in alphabetical order when appropriate by @dietmarw in https://github.com/OpenIPSL/OpenIPSL/pull/237
* Cleans up and proper place for base classes by @dietmarw in https://github.com/OpenIPSL/OpenIPSL/pull/239
* Fix gencls by @marcelofcastro in https://github.com/OpenIPSL/OpenIPSL/pull/240
* Add an initial speed deviation and remove superfluous initial equationns by @dietmarw in https://github.com/OpenIPSL/OpenIPSL/pull/241
* Cleans up the PSS and ES models  by @dietmarw in https://github.com/OpenIPSL/OpenIPSL/pull/242
* Clean up description strings wrt to units by @dietmarw in https://github.com/OpenIPSL/OpenIPSL/pull/243
* Clean up/reimplement FACTS models by @dietmarw in https://github.com/OpenIPSL/OpenIPSL/pull/244
* Add references to PSSE control models. by @dietmarw in https://github.com/OpenIPSL/OpenIPSL/pull/249
* Rescue of CGMES models by @dietmarw in https://github.com/OpenIPSL/OpenIPSL/pull/238
* Updated and fix the conversion script. by @dietmarw in https://github.com/OpenIPSL/OpenIPSL/pull/251
* Introduce nominal values to help numerical solvers by @dietmarw in https://github.com/OpenIPSL/OpenIPSL/pull/252
* Fix GGOV1 by @marcelofcastro in https://github.com/OpenIPSL/OpenIPSL/pull/254
* Fixing the initialisation issues with GENCLS and other PSS gen models by @dietmarw in https://github.com/OpenIPSL/OpenIPSL/pull/258
* Fix units by @marcelofcastro in https://github.com/OpenIPSL/OpenIPSL/pull/257
* Update Documentation in User's Guide by @marcelofcastro in https://github.com/OpenIPSL/OpenIPSL/pull/259
* Moving the CI files where they belong. by @dietmarw in https://github.com/OpenIPSL/OpenIPSL/pull/261
* Corrected display units of reactive power by @sergio-dorado in https://github.com/OpenIPSL/OpenIPSL/pull/268
* Improves the layout and documentation of the Plant model by @dietmarw in https://github.com/OpenIPSL/OpenIPSL/pull/271
* Converts CI from Travis to GitHub actions by @dietmarw in https://github.com/OpenIPSL/OpenIPSL/pull/272
* Improves documentation of GENCLS and unlocks parameters by @dietmarw in https://github.com/OpenIPSL/OpenIPSL/pull/273
* Clean up of NonElectrical/Logical by @dietmarw in https://github.com/OpenIPSL/OpenIPSL/pull/275
* Replaces custom LV and HV gate definitions with MSL blocks by @dietmarw in https://github.com/OpenIPSL/OpenIPSL/pull/274
* Changed `V` attribute to `v` (lowercase) in class `Bus` by @sergio-dorado in https://github.com/OpenIPSL/OpenIPSL/pull/279
* Clean-up and reimplementation of PowerFactory solar models by @tinrabuzin in https://github.com/OpenIPSL/OpenIPSL/pull/266
* Changing License by @marcelofcastro in https://github.com/OpenIPSL/OpenIPSL/pull/281
* Update README.md by @lvanfretti in https://github.com/OpenIPSL/OpenIPSL/pull/284
* Update CITATION.cff by @lvanfretti in https://github.com/OpenIPSL/OpenIPSL/pull/283
* Moves and converts Application Examples and old Examples to Examples and Tests, respectively by @marcelofcastro in https://github.com/OpenIPSL/OpenIPSL/pull/286
* Update the User's Guide by @dietmarw in https://github.com/OpenIPSL/OpenIPSL/pull/287
* Add option to use PadeDelay instead of normal delay to be used for linearisation by @dietmarw in https://github.com/OpenIPSL/OpenIPSL/pull/288
* Improved user interface for delay block in Turbine govenor by @dietmarw in https://github.com/OpenIPSL/OpenIPSL/pull/290
* Increase tests' and examples' compatibility with OMEdit by @marcelofcastro in https://github.com/OpenIPSL/OpenIPSL/pull/291
* Fixing soft PMU implementation by @marcelofcastro in https://github.com/OpenIPSL/OpenIPSL/pull/293
* Adding description sentence to all models in the library by @marcelofcastro in https://github.com/OpenIPSL/OpenIPSL/pull/294
* File clean up in preparation of 2.0.0 release by @dietmarw in https://github.com/OpenIPSL/OpenIPSL/pull/295
* Adding the release notes by @dietmarw in https://github.com/OpenIPSL/OpenIPSL/pull/297

### New Contributors
* @pierre-haessig made their first contribution in https://github.com/OpenIPSL/OpenIPSL/pull/156
* @BiswarupM made their first contribution in https://github.com/OpenIPSL/OpenIPSL/pull/169
* @ManuelNvro made their first contribution in https://github.com/OpenIPSL/OpenIPSL/pull/196
* @sergio-dorado made their first contribution in https://github.com/OpenIPSL/OpenIPSL/pull/268

**Full Changelog**: https://github.com/OpenIPSL/OpenIPSL/compare/v1.5.0...v2.0.0

<hr /><em>This discussion was created from the release <a href='https://github.com/OpenIPSL/OpenIPSL/releases/tag/v2.0.0'>v2.0.0</a>.</em>

**Comment by lvanfretti** [comment]
Thanks to @dietmarw and @marcelofcastro for the titanic effort in coordination and clean up, and to all the contributors weather specifically tagged or not, for bringing this new version to reality! Soon v3.0.0: [see issue here.](https://github.com/OpenIPSL/OpenIPSL/pull/299)
